stream-rpc/src/errors.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
looks like dup with json-rpc/types/src/errors.js

Reply:
it's not a 100% dup; the serialization of JsonRpcError is kinda similar, but the actual enums inside of them are different, and my usage of `ErrorData` is different as well. It'd be nice to standardize, but might be a bit hard given how strongly tied to VM codes the jsonrpc is vs the stream rpc

Reply:
Yeah I really think we should standardize the error here. I don't expect that it'll be too difficult to do. I was actually planning on cleaning up the Error type anyway and that should make it much easier to reuse.